main.js

```diff
@@ -2,26 +2,10 @@
 
 const axios = require('axios');
 
-const transform = (chain) => {
-  if (chain.evolves_to.length < 1) {
-    return {
-      name: chain.species.name,
-      variations: []
-    };
-  }
-  const evolutions = [];
-
-  // This is also covers a potential situation where there are
-  // multiple evolution chains
-  chain.evolves_to.forEach(chain => {
-    evolutions.push(transform(chain));
-  });
-
-  return {
-    name: chain.species.name,
-    variations: evolutions
-  };
-};
+const transform = (chain) => ({
+  name: chain?.species?.name,
+  variations: chain?.evolves_to?.map(transform)
+});
```

Comment:
would you be able to explain this one better for me?

Reply:
Transform is an arrow function that returns an object with 2 properties, `name` and `variations`.

The `=> ({` syntax means that the function will return an object rather than `=> {` which would initiate a code block. e.g.
```shell
$ node
Welcome to Node.js v16.18.1.
Type ".help" for more information.
> const fn = () => ({ foo: 'bar' })
undefined
> fn()
{ foo: 'bar' }
```

The value of `name` should be clear.

The value of `variations` is the result of mapping itself (`transform`) over the `evolves_to` property of the argument.

Reply:
Whoa, Daniel, I did know you could do this, it's so cool, thank you 😎